replica/table.cc

```diff
@@ -1119,6 +1119,7 @@ compaction_group::update_sstable_lists_on_off_strategy_completion(sstables::comp
     co_await _t.get_row_cache().invalidate(std::move(updater), std::move(empty_ranges));
     _t.get_row_cache().refresh_snapshot();
     _t.rebuild_statistics();
+    trigger_compaction();
```

Comment:
I think that the caller (`on_compaction_completion`) is in a better position to decide about that.
I'd leave this function focused on updating the lists.

BTW, I see that `update_main_sstable_list_on_compaction_completion` calls `delete_atomically` for `desc.old_sstables`. Who does that for the compacted maintenance sstables?

Reply:
> on_compaction_completion

Against my desire, the deletion lives in compaction_manager::offstrategy_compaction_task::run_offstrategy_compaction(). The reason for that is that input sstables aren't necessarily removed if some of them didn't have to go through reshape. E.g. input sstables for reshape conform to the strategy invariant to some extent.

But I can move it there for symmetry, which is what I always wanted to be honest. We can delete input sstables in update_main_sstable_list_on_compaction_completion() which aren't present in output set, meaning they won't be present anywhere later on. Looks like there's even room for further code reuse.

Reply:
Of course, that's work for another patch which will come in a different PR.